code/modules/robotics/bot/chefbot.dm

```diff
@@ -39,117 +42,297 @@
 /obj/machinery/bot/chefbot/proc/why_is_it_bad()
 	return pick("IS FUCKING [pick("RAW", "BLAND", "UNDERCOOKED", "OVERCOOKED", "INEDIBLE", "RANCID", "DISGUSTING")]", "LOOKS LIKE [pick("BABY VOMIT", "A MUSHY PIG'S ASS", "REGURGITATED DONKEY SHIT", "A PILE OF ROTTING FLIES", "REFINED CAT PISS")]")
 
+#define FOOD_QUALITY_HORSESHIT 0
+#define FOOD_QUALITY_SHIT 1
+#define FOOD_QUALITY_GOOD_SHIT 2
 /obj/machinery/bot/chefbot/proc/yell()
-	if (!src.emagged || prob(50))
-		var/obj/item/reagent_containers/food/snacks/shitfood
+	if (!src.emagged)
+		var/obj/item/reagent_containers/food/snacks/food_to_judge
 		var/mob/living/carbon/human/thechef
 		var/mob/dork
 		var/is_thechef_the_chef = 0
+		var/how_shit = FOOD_QUALITY_HORSESHIT //ITS ALL DONKEY PISS UNTIL I SAY OTHERWISE
 		for (var/obj/item/reagent_containers/food/snacks/probablyshitfood in view(7, src))
 			if (probablyshitfood in calledout)
 				continue
-			if (probablyshitfood.quality < 2)
-				shitfood = probablyshitfood
-				break
-		if (shitfood)
-			raging = 1
-			icon_state = "chefbot-mad"
-			for_by_tcl(M, /mob/living/carbon/human)
-				if(!IN_RANGE(M, src, 7))
-					continue
-				if (M.mind)
-					if (M.mind.assigned_role == "Chef")
-						thechef = M
-						is_thechef_the_chef = 1
-						break
-				if (M.wear_id)
-					if (findtext(M.wear_id:assignment, "chef") || findtext(M.wear_id:assignment, "cook"))
-						thechef = M
-						is_thechef_the_chef = 1
-						break
-				if (!thechef)
+			food_to_judge = probablyshitfood
+			calledout += food_to_judge //judgment is FINAL. We only judge once.
```

Comment:
permanent ref to the food preventing it from ever being garbage collected

Reply:
I now changed "calledout" to null upon exploding the bot. Though i'm not sure how garbage collecting works, so do tell me if i should instead empty the list at certain intervals or under certain conditions.

Reply:
I instead added a "is_judged" var on food/snacks to know if a food is judged, should avoid permanent refs.